torchgeo/datasets/usavars.py

```diff
@@ -0,0 +1,212 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""USAVars dataset."""
+
+import glob
+import os
+from typing import Callable, Dict, List, Optional
+
+import matplotlib.pyplot as plt
+import numpy as np
+import rasterio
+import torch
+from matplotlib.figure import Figure
+from torch import Tensor
+
+from .geo import VisionDataset
+from .utils import download_url, extract_archive
+
+
+class USAVars(VisionDataset):
+    """USAVars dataset.
+
+    Dataset format:
+    * images are 4-channel tifs
+    * labels are singular float values
+
+    Dataset labels:
+    - tree cover
+    - elevation
+    - population density
+    - nighttime lights
+    - income per houshold
+    - road length
+    - housing price
+
+    .. versionadded:: 0.3
+    """
+
+    url_prefix = (
+        "https://files.codeocean.com/files/verified/"
+        + "fa908bbc-11f9-4421-8bd3-72a4bf00427f_v2.0/data/int/applications"
+    )
+    pop_csv_suffix = "CONTUS_16_640_POP_100000_0.csv?download"
+    uar_csv_suffix = "CONTUS_16_640_UAR_100000_0.csv?download"
+
+    data_url = "https://mosaiks.blob.core.windows.net/datasets/uar.zip"
+    zipfile = "usavars.zip"
+
+    md5 = "677e89fd20e5dd0fe4d29b61827c2456"
+
+    label_urls = {
+        "housing": f"{url_prefix}/housing/outcomes_sampled_housing_{pop_csv_suffix}",
+        "income": f"{url_prefix}/income/outcomes_sampled_income_{pop_csv_suffix}",
+        "roads": f"{url_prefix}/roads/outcomes_sampled_roads_{pop_csv_suffix}",
+        "nightligths": f"{url_prefix}/nightlights/"
+        + f"outcomes_sampled_nightlights_{pop_csv_suffix}",
+        "population": f"{url_prefix}/population/"
+        + f"outcomes_sampled_population_{uar_csv_suffix}",
+        "elevation": f"{url_prefix}/elevation/"
+        + f"outcomes_sampled_elevation_{uar_csv_suffix}",
+        "treecover": f"{url_prefix}/treecover/"
+        + f"outcomes_sampled_treecover_{uar_csv_suffix}",
+    }
+
+    def __init__(
+        self,
+        root: str = "data",
+        transforms: Optional[Callable[[Dict[str, Tensor]], Dict[str, Tensor]]] = None,
```

Comment:
I think we could have an argument like `labels: Sequence[str] = self.ALL_LABELS` that lets users choose which of the labels to load. Then the dictionary returned by getitem can have a "labels" key that is a vector of values corresponding to the labels passed in.

E.g. `ds = USAVars(root="data/", labels=["population", "elevation"])`   would give me a dataset where `ds[0]` is a dictionary like:
```
{
  "image": ...,
  "labels": [population_value, elevation_value]
}
```